pandas/_libs/tslibs/period.pyx

```diff
@@ -2404,6 +2405,13 @@ class Period(_Period):
                 value = str(value)
             value = value.upper()
             dt, reso = parse_time_string(value, freq)
+            try:
```

Comment:
I think the correct fix is to have `parse_time_string` return `reso = 'N'`, and amend the dateutil parser to parse our nanosecond resolution.

The return may need to be:
`dt, reso, nanoseconds = parse_time_string(value, freq)`

Reply:
Agree, 
The thing is `parse_time_sting` calls `parse_datetime_string_with_reso` then  `dateutil_parse` which finally calls `dateutil.parser.DEFAULTPARSER._parse` that does not returns nanosecond values.
So, is is a `pandas` bug or a `dateutil` enhancement ?

Reply:
More of a pandas bug/enhancement. I don't think this change belongs in `dateutil` because it primarily operates on `datetime.datetime` objects that only support up to microsecond resolution.

So we may need to port over the dateutil parser and include logic to recognize nanoseconds

Reply:
Thanks for the responses. As matteosantama suggested, https://github.com/pandas-dev/pandas/issues/34621#issuecomment-640996489, I've tried to use the `Timestamp` constructor + I've added tests + the `try except` was modified.
The point is the Timestamp does not returns the `freq` variable that is used by the Period constructor : we need to keep the `parse_time_sting` calls. This is not a nice solution.

Reply:
Yeah instead of calling the `Timestamp` constructor you may be able to borrow from our the Timestamp's date parser `parse_iso_8601_datetime`

Reply:
@mroeschke : the last PR is green, with tests inside. However, I used the `Timestamp` constructor.

I tried to use `parse_iso_8601_datetime`, but was not successful. I had an error saying the sources are not compiled. Is there a ressource to learn how to use C function ? I'd like to have a look to this.

Reply:
Actually I don't think using the `parse_iso_8601_datetime` would be as straightforward in this case. Using `Timestamp` as you have it might be a good solution in the interim.